### PR TITLE
Марковников Никита, 2537

### PR DESCRIPTION
**Pull request by mikita95**

### CODE REVIEW

File: app/build.gradle [hunk 1]
```diff
@@ -17,4 +24,8 @@ android {
             proguardFiles getDefaultProguardFile('proguard-android.txt'), 'proguard-rules.pro'
         }
     }
+    productFlavors {
+    }
+}
+dependencies {
 }
```
Comment (thevery) [inline app/build.gradle:31]: I don't quite understand why do you have almost identical (an useless!) changes with https://github.com/IFMO-MobDev-2014/lesson1/pull/4/files#diff-1

Comment (mikita95) [inline app/build.gradle:31]: I don't know too. I couldn't run the project without them. Is it bad?

Comment (thevery) [inline app/build.gradle:31]: just run the debug build - it should work without theese changes

Comment (mikita95) [inline app/build.gradle:31]: Ok. I'll try it in future. Thank you.

File: app/src/main/java/ru/ifmo/md/lesson1/MyActivity.java [hunk 2]
```diff
@@ -11,6 +13,8 @@
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+        requestWindowFeature(Window.FEATURE_NO_TITLE);
+        getWindow().setFlags(WindowManager.LayoutParams.FLAG_FULLSCREEN, WindowManager.LayoutParams.FLAG_FULLSCREEN);
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/MyActivity.java:17]: not required and will not work on android 4.4+ correctly anyway

**Comment by thevery** [comment]
ok = 8
~30 FPS

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 3]
```diff
@@ -47,67 +49,73 @@ public void run() {
...
     void updateField() {
-        int[][] field2 = new int[width][height];
-        for (int x=0; x<width; x++) {
-            for (int y=0; y<height; y++) {
-
-                field2[x][y] = field[x][y];
-
-                for (int dx=-1; dx<=1; dx++) {
-                    for (int dy=-1; dy<=1; dy++) {
+        boolean flag;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:85]: bad naming

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 4]
```diff
@@ -47,67 +49,73 @@ public void run() {
...
+                            myField[x + y * width] = palette[field[z][x2][y2]];
+                            flag = false;
+                            break;
                         }
                     }
                 }
             }
         }
-        field = field2;
+        z ^= 1;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:112]: bitwise magic makes your code harder to understand

File: app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java [hunk 5]
```diff
@@ -1,25 +1,26 @@
...
 
-/**
-* Created by thevery on 11/09/14.
-*/
 class WhirlView extends SurfaceView implements Runnable {
-    int [][] field = null;
-    int width = 0;
-    int height = 0;
-    int scale = 4;
+    int[][][] field = null;
```
Comment (thevery) [inline app/src/main/java/ru/ifmo/md/lesson1/WhirlView.java:13]: two 2D arrays access should be faster than using 3D array

**Pull request closed without merging**